docs/developer/contributing/api-guidelines.rst

```diff
@@ -204,31 +204,40 @@ each API endpoint.
 
 A generic ``sort`` parameter **must** be used to describe sorting rules. If
 there are multiple fields that a collection can be sorted by, the ``sort``
-parameter **must** accept a list of comma separated fields, indicating
-descending sort priority. A minus sign implies descending sort order in the
-subsequent field.
+parameter **must** accept a list of comma separated fields. The format of the
+``sort`` parameter values has to follow: ``field:ASC|DESC``.
 
 - Retrieving all files, ordered alphabetically by file type and by descending
   file size:
-  ``GET /files?sort=type,-size``
+  ``GET /files?sort=type:ASC,size:DESC``
 
 6.3 Pagination
 ~~~~~~~~~~~~~~
 
 Pagination **must** be supported by all endpoints returning more than just
-a single resource. Pagination **must** implemented using the query
-parameters ``page`` and ``per_page``. Link headers **must** be included in the
-response indicating:
+a single resource. Pagination **must** be implemented using the query
+parameters ``page`` and ``per_page``. Both the parameters have to accept
+any positive integer value. The ``page`` parameter **must** default to 1 
+while the default for the ``per_page`` should be set to some value reasonable 
+for an endpoint. Non-positive values for both parameters causes 
+endpoint to return ``BAD_REQUEST (400)`` response.
+
+Link headers **must** be included in the response indicating:
 
 - first page ``rel="first"``;
-- previous page ``rel="prev"``;
-- current page ``rel="self"``;
-- next page ``rel="next"``;
+- previous page ``rel="prev"`` (present only when the current page is not the first page);
+- next page ``rel="next"`` (present only when the current page is not the last page);
 - last page ``rel="last"``.
 
 Link headers are described by `RFC 8288
 <https://tools.ietf.org/html/rfc8288>`_.
 
-In cases where the total number of resources is not known in advance, the
-parameters ``limit`` and ``offset`` **should** be used instead. In this case,
-the link for the last page is omitted in the response header.
+Additional headers **must** be included in the response to present information
+about pagination:
+
+- ``Total``	for the total number of items;
```

Comment:
Oh, and another thing (sorry for doing this one-by-one). It is fine for me to always require the pagination headers, even when the link headers are present. But I think the case where the total number of result is not known is actually realistic, so it should be possible to omit this header. 
For example, a call to `https://gitlab.com/api/v4/projects?page=2&per_page=10` won't give you the total either number of pages either.